examples/datadog/main.tf

```diff
@@ -0,0 +1,197 @@
+    # This can be restricted further to specific port based on the requirement for each Add-on e.g., metrics-server 4443, spark-operator 8080, karpenter 8443 etc.
+    # Change this according to your security requirements if needed
+    ingress_cluster_to_node_all_traffic = {
+      description                   = "Cluster API to Nodegroup all traffic"
+      protocol                      = "-1"
+      from_port                     = 0
+      to_port                       = 0
+      type                          = "ingress"
+      source_cluster_security_group = true
+    }
+  }
+
+  managed_node_groups = {
+    mg_5 = {
+      node_group_name      = "managed-ondemand"
+      instance_types       = ["m5.large"]
+      subnet_ids           = module.vpc.private_subnets
+      force_update_version = true
+    }
+  }
+
+  tags = local.tags
+}
+
+module "eks_blueprints_kubernetes_addons" {
+  source = "../../modules/kubernetes-addons"
+
+  eks_cluster_id               = module.eks_blueprints.eks_cluster_id
+  eks_cluster_endpoint         = module.eks_blueprints.eks_cluster_endpoint
+  eks_oidc_provider            = module.eks_blueprints.oidc_provider
+  eks_cluster_version          = module.eks_blueprints.eks_cluster_version
+  eks_worker_security_group_id = module.eks_blueprints.worker_node_security_group_id
+
+  enable_datadog_operator = true
+
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# Supporting Resources
+#---------------------------------------------------------------
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 3.0"
+
+  name = local.name
+  cidr = local.vpc_cidr
+
+  azs             = local.azs
+  public_subnets  = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k)]
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k + 10)]
+
+  enable_nat_gateway   = true
+  single_nat_gateway   = true
+  enable_dns_hostnames = true
+
+  # Manage so we can name
+  manage_default_network_acl    = true
+  default_network_acl_tags      = { Name = "${local.name}-default" }
+  manage_default_route_table    = true
+  default_route_table_tags      = { Name = "${local.name}-default" }
+  manage_default_security_group = true
+  default_security_group_tags   = { Name = "${local.name}-default" }
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.name}" = "shared"
+    "kubernetes.io/role/elb"              = 1
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${local.name}" = "shared"
+    "kubernetes.io/role/internal-elb"     = 1
+  }
+
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# Datadog agent
+#---------------------------------------------------------------
+
+resource "kubernetes_secret_v1" "datadog_api_key" {
```

Comment:
the secret and the DatadogAgent manifest should be moved to the addon itself

Reply:
My only concern with that, is having this _secret_ API key in the Terraform state. I know for a fact it would be undesirable at my company.  

If I were to do this, I'd store the secret in secrets manager or parameter store, and sync it to the cluster.

Reply:
This is the reason why I included the agent configuration as an example and not in the addon itself.

Reply:
completely agree - but we can provide users the ability to create a secret, or reference a secret. either way a secret has to exist, but we can provide options. the whole premise is that an addon *should be installed with a single flag, but here we are saying users have to install and then semi-manually go and do some additional work

Reply:
I was viewing the addon as "the operator", but having it handle the agent creation would definitely be preferable. I mean tbh, I wasn't totally sure about dragging the not-really-official kubectl provider into the main code base.

Certainly an optional secret is possible and what I was considering.

Reply:
lets see what the others think - @askulkarni2 / @vara-bonthu / @Zvikan

Reply:
apologies for the delay @bobdoah - after considering both avenues, I think you were correct with your original approach. This is the current practice for CRDs so we will continue with this approach